### ISSUE TITLE
Responsive Rules not working on resize

### ISSUE THREAD

**Issue by seabass223** [amcharts/amcharts5#151]
The demo chart at the bottom of this page does not work either.  The rule runs once at the beginning but never triggers the _relevant_ _applying_ or _removing_ functions again during a resize.     "@amcharts/amcharts5": "^5.0.17",
https://www.amcharts.com/docs/v5/concepts/responsive/

**Comment by seabass223** [comment]
after some investigation I found that in ResponsiveTheme.js, the addRule method calls this._applyRule(rule) the first time, and any updates are left to be fired when the root container changes width or height, but those events are never fired from the container.
```
this._root._rootContainer.on("width", function (_width) {
  //never fires
  if (_this._isUsed()) {
    _this._applyRule(rule);
  }
}),
  this._root._rootContainer.on("height", function (_height) {
    //never fires
    if (_this._isUsed()) {
      _this._applyRule(rule);
    }
  });
```
as a hack/temporary fix, setting the values to their current value forces the above events to fire.

```
  chart.events.on('boundschanged', (e) => {
      root._rootContainer.set('height', root._rootContainer.height());
      root._rootContainer.set('width', root._rootContainer.width());
    });
```

**Comment by martynasma** [comment]
Fixed in [5.0.18](https://github.com/amcharts/amcharts5/releases/tag/5.0.18).

## [5.0.18] - 2021-12-19

### Changed
- When auto-wrapping labels, text will no longer break on dots to prevent fractional numbers from wrapping.
- Scrollable `Container`'s scrollbar will auto-hide if contents fit in it without scrolling.

### Fixed
- Disabling click-toggling of legend, was disabling hover events, too.
- `ResponsiveTheme`'s custom rules were not working properly.
- `Exporting` was not applying proper background color to exported images.
- `CategoryDateAxis` grid and labels placing algorithm improved.
- Sometimes bullets of XYChart were not being properly clipped by the bounds of the plot area.
- Memory leak related to: https://bugs.chromium.org/p/chromium/issues/detail?id=1279394 was fixed.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
